src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -317,12 +317,27 @@ def purge(self, service_name: str) -> None:
             '--namespace', cast(str, spec.service_id),
             'rm', 'grace',
         ]
-        subprocess.run(
-            cmd,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            timeout=10
-        )
+        try:
+            result = subprocess.run(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                timeout=10
+            )
+        except Exception as ex:
+            err_msg = (
+                f'Command {cmd} failed while trying to remove grace file for nfs '
+                f'service nfs.{spec.service_id}: {str(ex)}'
+            )
+            logger.warning(err_msg)
+            raise OrchestratorError(err_msg)
+        if result.returncode:
```

Comment:
You can combine this with above by setting `check=True` for subprocess.run. Otherwise, I wouldn't bother with the above try-except.